aurora-locker/src/Locker.sol

```diff
@@ -76,7 +91,7 @@ contract Locker {
         );
 
         // Combine the two promises into a single promise and schedule it.
-        mintOnNear.then(callback).transact();
+        mintOnNear.then(callback).lazy_transact();
```

Comment:
Yes, I think it is better to separate it in two calls rather than having lazy execution here. Otherwise, we need to setup an off-chain mechanism to finalize all transfers.

Reply:
I separated the creation from deposit in #18 , but still it fails due to running out of gas when I try the eager execution instead of delayed. I'll have to look into it more.

Reply:
Can we add an assertion about the amount of gas used to know the latest numbers? Probably this would be good to have per different method.

Reply:
Turned out the issue was in the engine itself: https://github.com/aurora-is-near/aurora-engine/pull/622